src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -86,15 +160,103 @@ public function Accept(\IMapperVisitorRestricted $visitor, $cachingEnabled, \IMa
+                'activeNodeId' => $this->activeNodeId,
+                'rootTreeId' => $rootTreeId,
+                'fieldName' => $this->fieldName,
+                'portalSelectMode' => $portalSelectMode,
+            ],
+            PATH_CMS_CONTROLLER.'?',
+            '&'
+        );
+        $visitor->SetMappedValue('treeNodesAjaxUrl', $url);
+
+        $url = $this->urlUtil->getArrayAsUrl(
+            [
+                'pagedef' => $pagedef,
+                'module_fnc' =>
+                    [
+                        'contentmodule' => 'ExecuteAjaxCall'
+                    ],
+                '_fnc' => 'updateSelection',
+                'table' => $tableName,
+                'currentRecordId' => $currentRecordId,
+                'fieldName' => $this->fieldName
+            ],
+            PATH_CMS_CONTROLLER.'?',
+            '&'
+        );
+        $visitor->SetMappedValue('updateSelectionUrl', $url);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function DefineInterface()
+    {
+        parent::DefineInterface();
+        $externalFunctions =
+            [
+                'getTreeNodes',
+                'updateSelection'
+            ];
+        $this->methodCallAllowed = array_merge($this->methodCallAllowed, $externalFunctions);
+    }
+
+    /**
+     * Is called via ajax.
+     * Returns requested treeNode or rootTreeNode (id = "#") with all its children.
+     * The return value is converted to JSON with array brackets around it, jstree.js needs it that way
+     */
+    protected function getTreeNodes(): array
+    {
+        $this->fieldName = $this->inputFilterUtil->getFilteredGetInput('fieldName', '');
+        $portalSelectMode = $this->inputFilterUtil->getFilteredGetInput('portalSelectMode', '');
+        $this->isPortalSelectMode = $portalSelectMode === 'portalSelect' ? true : false;
+        $this->isPortalHomeNodeSelectMode = $portalSelectMode === 'portalHomePage' ? true : false;  //also 404-page-selection
+
+        $this->activeNodeId = $this->inputFilterUtil->getFilteredGetInput('activeNodeId', '');
+        $rootTreeId = $this->inputFilterUtil->getFilteredGetInput('rootTreeId', '');
+        if ('' === $rootTreeId) {
+            return [];
+        }
+        $rootNode = new \TdbCmsTree();
+        $rootNode->SetLanguage(\TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+        $rootNode->Load($rootTreeId);
+
+        $treeData[] = $this->createTreeDataModel($rootNode, 0);
+        return $treeData;
+    }
+
+    /**
+     * Is called via ajax.
+     */
+    protected function updateSelection(): ?string
+    {
+        $tableName = $this->inputFilterUtil->getFilteredGetInput('table', '');
+        $nodeId = $this->inputFilterUtil->getFilteredGetInput('nodeId', '');
+        $currentRecordId = $this->inputFilterUtil->getFilteredGetInput('currentRecordId', '');
+        $fieldName = $this->inputFilterUtil->getFilteredGetInput('fieldName', '');
```

Comment:
This does not overwrite the new instance variable?

Reply:
I've removed the instance variable fieldName. See above.